### ISSUE TITLE
RSolr::Document.add_field is not compatible with atomic updates in 2.0.0.pre1

### ISSUE THREAD

**Issue by serggl** [rsolr/rsolr#165]
Correction: It works fine for storing new value, e.g:
```ruby
document.add_field(:some_field, 'some_value', {update: :set})
```

BUT if you wan to clear this value (as per [Solr wiki](https://wiki.apache.org/solr/UpdateXmlMessages#Examples_of_adding_docs_with_various_optional_attributes)), e.g:
```ruby
document.add_field(:some_field, nil, {update: :set, null: true})
```

It wont do anything with a resulting document because of these lines https://github.com/rsolr/rsolr/blob/master/lib/rsolr/document.rb#L40-L41

P.S. this issue causes Travis build to fail to this Sunspot PR: https://github.com/sunspot/sunspot/pull/826

**Comment by cbeer** [comment]
Thanks for the report.

I wonder why RSolr rejects empty values in the first place. The current behavior originally came from https://github.com/rsolr/rsolr/blob/v1.1.2/lib/rsolr/xml.rb#L21 (unhelpfully introduced in https://github.com/rsolr/rsolr/commit/b7033c0e0e219f526a82b5c68cd217ce462b3b25).

With this major release, maybe this is a good opportunity to reconsider. Thoughts, @rsolr/core and others?

**Comment by serggl** [comment]
@cbeer I believe this behaviour is not correct. For instance you wont be able to clear value for some multi-valued stored field, because this line will filter it out
BTW initialiser you mentioned in out of sync with the `add_field` method below. So that is why it worked properly for sunspot gem before

**Comment by cbeer** [comment]
It looks like we fixed this in #167.
